# Bonus: Temperature Analysis I

In [45]:
import pandas as pd
from datetime import datetime as dt
from scipy import stats
from numpy import mean

In [46]:
# "tobs" is "temperature observations"
df = pd.read_csv('Resources/hawaii_measurements.csv')
df.head()

station        date  prcp  tobs
0  USC00519397  2010-01-01  0.08    65
1  USC00519397  2010-01-02  0.00    63
2  USC00519397  2010-01-03  0.00    74
3  USC00519397  2010-01-04  0.00    76
4  USC00519397  2010-01-06   NaN    73

In [47]:
# Convert the date column format from string to datetime
df['date']= pd.to_datetime(df['date'])
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 19550 entries, 0 to 19549
Data columns (total 4 columns):
 #   Column   Non-Null Count  Dtype         
---  ------   --------------  -----         
 0   station  19550 non-null  object        
 1   date     19550 non-null  datetime64[ns]
 2   prcp     18103 non-null  float64       
 3   tobs     19550 non-null  int64         
dtypes: datetime64[ns](1), float64(1), int64(1), object(1)
memory usage: 611.1+ KB


In [48]:
# Set the date column as the DataFrame index
df_index=df.set_index("date")
df_index.head()

station  prcp  tobs
date                               
2010-01-01  USC00519397  0.08    65
2010-01-02  USC00519397  0.00    63
2010-01-03  USC00519397  0.00    74
2010-01-04  USC00519397  0.00    76
2010-01-06  USC00519397   NaN    73

In [49]:
# Drop the date column
df_drop= df.drop(columns=['date'])
df_drop

station  prcp  tobs
0      USC00519397  0.08    65
1      USC00519397  0.00    63
2      USC00519397  0.00    74
3      USC00519397  0.00    76
4      USC00519397   NaN    73
...            ...   ...   ...
19545  USC00516128  0.09    71
19546  USC00516128   NaN    78
19547  USC00516128  0.56    76
19548  USC00516128  0.50    76
19549  USC00516128  0.45    76

[19550 rows x 3 columns]

### Compare June and December data across all years 

In [50]:
from scipy import stats

In [51]:
# Filter data for desired months
# Fraud_df.at[:,'trans_date_trans_time']=pd.to_datetime(Fraud_df['trans_date_trans_time'])
# Fraud_df.at[:,'hour']=Fraud_df['trans_date_trans_time'].dt.hour
df.at[:,'month']=df['date'].dt.month
df


station       date  prcp  tobs  month
0      USC00519397 2010-01-01  0.08    65      1
1      USC00519397 2010-01-02  0.00    63      1
2      USC00519397 2010-01-03  0.00    74      1
3      USC00519397 2010-01-04  0.00    76      1
4      USC00519397 2010-01-06   NaN    73      1
...            ...        ...   ...   ...    ...
19545  USC00516128 2017-08-19  0.09    71      8
19546  USC00516128 2017-08-20   NaN    78      8
19547  USC00516128 2017-08-21  0.56    76      8
19548  USC00516128 2017-08-22  0.50    76      8
19549  USC00516128 2017-08-23  0.45    76      8

[19550 rows x 5 columns]

In [60]:
df_June=df.loc[df["month"]==6]
df_June

station       date  prcp  tobs  month
133    USC00519397 2010-06-01  0.00    78      6
134    USC00519397 2010-06-02  0.01    76      6
135    USC00519397 2010-06-03  0.00    78      6
136    USC00519397 2010-06-04  0.00    76      6
137    USC00519397 2010-06-05  0.00    77      6
...            ...        ...   ...   ...    ...
19492  USC00516128 2017-06-26  0.02    79      6
19493  USC00516128 2017-06-27  0.10    74      6
19494  USC00516128 2017-06-28  0.02    74      6
19495  USC00516128 2017-06-29  0.04    76      6
19496  USC00516128 2017-06-30  0.20    75      6

[1700 rows x 5 columns]

In [53]:
# Filter data for desired months
df_Dec=df.loc[df["month"]==12]
df_Dec

station       date  prcp  tobs  month
305    USC00519397 2010-12-01  0.04    76     12
306    USC00519397 2010-12-03  0.00    74     12
307    USC00519397 2010-12-04  0.00    74     12
308    USC00519397 2010-12-06  0.00    64     12
309    USC00519397 2010-12-07  0.00    64     12
...            ...        ...   ...   ...    ...
19323  USC00516128 2016-12-27  0.14    71     12
19324  USC00516128 2016-12-28  0.14    71     12
19325  USC00516128 2016-12-29  1.03    69     12
19326  USC00516128 2016-12-30  2.37    65     12
19327  USC00516128 2016-12-31  0.90    65     12

[1517 rows x 5 columns]

In [54]:
# Identify the average temperature for June
mean_June=df_June['tobs'].mean()
mean_June

74.94411764705882

In [55]:
# Identify the average temperature for December
mean_Dec=df_Dec['tobs'].mean()
mean_Dec

71.04152933421226

In [58]:
# Create collections of temperature data
June_temp=df_June['tobs']
Dec_temp=df_Dec['tobs']

In [1]:
# Run paired t-test
# stats.ttest_rel(June_temp,Dec_temp)

In [61]:
#unpaired(indepent)t-test
stats.ttest_ind(June_temp,Dec_temp)

Ttest_indResult(statistic=31.60372399000329, pvalue=3.9025129038616655e-191)

### Analysis

####  The t test tells you how significant the differences between group means are. It lets you know if those differences in means could have happened by chance. A paired t-test is designed to compare the means of the same group or item under two separate scenarios. An unpaired t-test compares the means of two independent or unrelated groups. 
#### In this analysis, unpaired t-test is used because the means of two independent groups are compared. The temprature in June and December are the data of two independent groups. The pvalue of 3.90 is higher than 0.05, and indicated the result is non-significant.